gtsfm/ui/dot_graph_generator.py

```diff
@@ -0,0 +1,137 @@
+        print("!\n" * 100)
+        print(RegistryHolder.get_registry())
+
+        seen_metadata = set()
+        for cls_name, cls_type in RegistryHolder.get_registry().items():
+            # don't add the base class to the graph
+            if cls_name == "GTSFMProcess":
+                continue
+
+            # don't add any test classes to the graph, unless in testing mode
+            if not self._test_mode and cls_name.startswith("Fake"):
+                continue
+
+            # get UI metadata of class
+            metadata = cls_type.get_ui_metadata()
+
+            # skip duplicates
+            # happens when concrete classes implement an abstract class without overwriting get_ui_metadata()
+            if metadata in seen_metadata:
+                continue
+
+            seen_metadata.add(metadata)
+
+            self._add_metadata_to_graph(metadata)
+
+    def _add_metadata_to_graph(self, metadata: UiMetadata):
+        """Given the UiMetadata from a GTSFMProcess, clean it up, then add to graph."""
+
+        display_name = metadata.display_name
+
+        # autocast strings to one-element tuples
+        #
+        # a common error is to define a one-element tuple like so:
+        # >>> ("Input Product")
+        # but Python auto-casts this to a raw str
+        input_products = metadata.input_products
+        if type(input_products) == str:
+            input_products = (input_products,)
+
+        output_products = metadata.output_products
+        if type(output_products) == str:
+            output_products = (output_products,)
+
+        parent_plate = metadata.parent_plate
+
+        # add cleaned-up metadata to graph
+        self._add_nodes_and_edges(display_name, input_products, output_products, parent_plate)
+
+    def _add_nodes_and_edges(
+        self, display_name: str, input_products: Tuple[str], output_products: Tuple[str], parent_plate: str
+    ):
+        """Given clean UI metadata, add blue/gray nodes to the graph."""
+
+        style = self._style
+
+        # add process, all products to graph as Nodes
+        self._graph.add_node(
+            pydot.Node(display_name, shape=style.node_shape, style=style.node_style, fillcolor=style.blue_fillcolor)
+        )
+        for product_name in input_products + output_products:
+            self._graph.add_node(
+                pydot.Node(product_name, shape=style.node_shape, style=style.node_style, fillcolor=style.gray_fillcolor)
+            )
+
+        # add Edges for all input_products -> blue node -> all output_products
+        for input_product_name in input_products:
+            self._graph.add_edge(pydot.Edge(input_product_name, display_name, color=style.arrow_color))
+        for output_product_name in output_products:
+            self._graph.add_edge(pydot.Edge(display_name, output_product_name, color=style.arrow_color))
+
+    def save_graph(self, filepath=os.path.join(JS_ROOT, "src", "ui", "dot_graph_output.svg")):
```

Comment:
let's make the default arg a global var, like
```python
DEFAULT_GRAPH_VIZ_OUTPUT_PATH = JS_ROOT / "src" / "ui" / "dot_graph_output.svg"
```
and then using it here as def save_graph(self, filepath=DEFAULT_GRAPH_VIZ_OUTPUT_PATH) -> None`?